exercises/zipper/canonical-data.json

```diff
@@ -0,0 +1,715 @@
+              }
+            }
+          }
+        },
+        {
+          "description": "set_left with leaf",
+          "property": "property",
+          "initialTree": {
+            "value": 1,
+            "left": {
+              "value": 2,
+              "left": null,
+              "right": {
+                "value": 3,
+                "left": null,
+                "right": null
+              }
+            },
+            "right": {
+              "value": 4,
+              "left": null,
+              "right": null
+            }
+          },
+          "operations": [
+            {
+              "operation": "from_tree"
+            },
+            {
+              "operation": "left"
+            },
+            {
+              "operation": "set_left",
+              "item": {
+                "value": 5,
+                "left": null,
+                "right": null
+              }
+            },
+            {
+              "operation": "to_tree"
+            }
+          ],
+          "expected": {
+            "type": "tree",
+            "value": {
+              "value": 1,
+              "left": {
+                "value": 2,
+                "left": {
+                  "value": 5,
+                  "left": null,
+                  "right": null
+                },
+                "right": {
+                  "value": 3,
+                  "left": null,
+                  "right": null
+                }
+              },
+              "right": {
+                "value": 4,
+                "left": null,
+                "right": null
+              }
+            }
+          }
+        },
+        {
+          "description": "set_right with null",
+          "property": "property",
+          "initialTree": {
+            "value": 1,
+            "left": {
+              "value": 2,
+              "left": null,
+              "right": {
+                "value": 3,
+                "left": null,
+                "right": null
+              }
+            },
+            "right": {
+              "value": 4,
+              "left": null,
+              "right": null
+            }
+          },
+          "operations": [
+            {
+              "operation": "from_tree"
+            },
+            {
+              "operation": "left"
+            },
+            {
+              "operation": "set_right",
+              "item": null
+            },
+            {
+              "operation": "to_tree"
+            }
+          ],
+          "expected": {
+            "type": "tree",
+            "value": {
+              "value": 1,
+              "left": {
+                "value": 2,
+                "left": null,
+                "right": null
+              },
+              "right": {
+                "value": 4,
+                "left": null,
+                "right": null
+              }
+            }
+          }
+        },
+        {
+          "description": "set_right with subtree",
```

Comment:
"set_right with subtree" case currently needs a `to_tree` as its last operation, otherwise the `operations` result in a zipper, but the expected value is a tree.